### ISSUE TITLE
`hash(a::AbstractArray)` assumes that `first(LinearIndices(a)) == 1`

### ISSUE THREAD

**Issue by goretkin** [JuliaLang/julia#35597]
```julia
julia> hash(Base.IdentityUnitRange(2:4))
ERROR: BoundsError: attempt to access 3-element LinearIndices{1,Tuple{Base.IdentityUnitRange{UnitRange{Int64}}}} with indices 2:4 at index [1]
Stacktrace:
 [1] throw_boundserror(::LinearIndices{1,Tuple{Base.IdentityUnitRange{UnitRange{Int64}}}}, ::Tuple{Int64}) at ./abstractarray.jl:538
 [2] checkbounds at ./abstractarray.jl:503 [inlined]
 [3] getindex at ./indices.jl:448 [inlined]
 [4] hash(::Base.IdentityUnitRange{UnitRange{Int64}}, ::UInt64) at ./abstractarray.jl:2210
 [5] hash(::Base.IdentityUnitRange{UnitRange{Int64}}) at ./hashing.jl:18
 [6] top-level scope at REPL[3]:1
```

**Comment by goretkin** [comment]
(This means you cannot hash ndims=1 `OffsetArray`s.)

**Comment by Peiffap** [comment]
@goretkin Should this issue be closed? The relevant PRs (#35598, #35624) have been merged already.

**Comment by goretkin** [comment]
Looks like it!
